src/test/java/com/fasterxml/jackson/databind/deser/creators/DelegatingCreatorImplicitNames2543Test.java

```diff
@@ -0,0 +1,66 @@
+package com.fasterxml.jackson.databind.deser.creators;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.introspect.*;
+import org.junit.Test;
+
+import static com.fasterxml.jackson.annotation.JsonCreator.Mode.DELEGATING;
+import static com.fasterxml.jackson.annotation.JsonCreator.Mode.PROPERTIES;
+import static com.fasterxml.jackson.databind.testutil.DatabindTestUtil.a2q;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+public class DelegatingCreatorImplicitNames2543Test {
+    static class Data {
+
+        final String part1;
+        final String part2;
+
+        // this creator is considered a source of settable bean properties,
+        // used during deserialization
+        @JsonCreator(mode = PROPERTIES)
+        public Data(@JsonProperty("part1") String part1,
+                    @JsonProperty("part2") String part2) {
+            this.part1 = part1;
+            this.part2 = part2;
+        }
+
+        // no properties should be collected from this creator,
+        // even though it has an argument with an implicit name
+        @JsonCreator(mode = DELEGATING)
+        public static Data fromFullData(String fullData) {
+            String[] parts = fullData.split("\\s+", 2);
+            return new Data(parts[0], parts[1]);
+        }
+    }
+
+    private static class DelegatingCreatorNamedArgumentIntrospector
+            extends JacksonAnnotationIntrospector {
+        public String findImplicitPropertyName(AnnotatedMember member) {
+            if (member instanceof AnnotatedParameter p) {
+                if (p.getOwner() instanceof AnnotatedMethod m) {
+                    if (requireNonNull(m.getAnnotation(JsonCreator.class)).mode() == DELEGATING)
+                        return "fullData";
+                }
+            }
+            return super.findImplicitPropertyName(member);
+        }
+    }
+
+    private static final ObjectMapper MAPPER =  new ObjectMapper()
+            .setAnnotationIntrospector(new DelegatingCreatorNamedArgumentIntrospector());
+
+    @Test
+    public void testDeserialization() {
+        assertThatNoException()
+                .isThrownBy(() -> MAPPER.readValue(a2q("{'part1':'a','part2':'b'}"), Data.class));
```

Comment:
Could you run assertions on the deserialized data?

Reply:
Why use `assertThatNoException` instead of just letting exception be thrown, which will fail test.

Reply:
IMHO, `assertThatNoException` is more expressive (than a simple call) in telling what it is we are testing for. But I'm OK with getting rid of it, if you say so

Reply:
Done